packages/debug-test/DebuggingFeatures.test.ts

```diff
@@ -0,0 +1,236 @@
+
+const metro = new Metro();
+
+beforeAll(async () => {
+	metro.start();
+});
+
+afterAll(() => {
+	metro.stop();
+});
+
+// beforeEach(() => {});
+// afterEach(() => {});
+
+function skip_test(msg, fn){}
+
+test("debug target properties", async () => {
+	testLog.message(`test debug target properties`);
+
+	const settings = new PlaygroundDebugSettings({webDebugger: false, directDebugging: true, jsEngine: "Hermes"});
+	await settings.initialize();
+	try {
+		await selectPackage("Samples\\debugTest01");
+
+		const options : http.RequestOptions = {
+			hostname: 'localhost',
+			port: 8081,
+			path: '/json',
+			method: 'GET'
+		};
+
+		let request = null;
+		const dataPromise = new Promise((resolve, reject) => {
+			request = http.request(options, response => response.on('data', data => resolve(data)));
+		});
+
+		request.end();
+
+		const data = await dataPromise;
+		expect(Buffer.isBuffer(data)).toBeTruthy();
+		const debugTargets = JSON.parse(data.toString());
+		expect(Array.isArray(debugTargets)).toBeTruthy();
+		expect(debugTargets.length).toBeGreaterThanOrEqual(1);
+		for (const debugTarget of debugTargets) {
+			expect(debugTarget.hasOwnProperty('id')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('title')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('faviconUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('devtoolsFrontendUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('type')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('webSocketDebuggerUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('vm')).toBeTruthy();
+		}
+	}
+	finally {
+		await settings.uninitialize();
+	}
+});
+
+skip_test("debugger connection", async () => {
+
+	const settings = new PlaygroundDebugSettings({webDebugger: false, directDebugging: true, jsEngine: "Hermes"});
+	await settings.initialize();
+	try {
+		await selectPackage("Samples\\debugTest01");
+
+		const debugTargets = await getDebugTargets();
+		expect(debugTargets.length).toBe(2);
+
+		// testLog.message(`debugTarget0: URL: "${debugTargets[0].webSocketDebuggerUrl}", VM: "${debugTargets[0].vm}"`);
+		// testLog.message(`debugTarget1: URL: "${debugTargets[1].webSocketDebuggerUrl}", VM: "${debugTargets[1].vm}"`);
+		const ws = new WebSocket(debugTargets[0].webSocketDebuggerUrl);
+
+		ws.on('open', () => {
+			testLog.message(`WS on-open: sending Debugger.enable`);
+			ws.send(`{ "id":2, "method":"Debugger.enable", "params":{ "maxScriptsCacheSize":10000000 } }`)
+			testLog.message(`WS on-open: sending Debugger.pause`);
+			ws.send(`{ "id":11, "method":"Debugger.pause", "params":{}}`);
+		});
+
+
+		// 	{ "id":2, "method":"Debugger.enable", "params":{ "maxScriptsCacheSize":10000000 } }
+
+		ws.on('close', () => { testLog.message(`WS on-close`);});
+
+		ws.on('message', (data) => {
+			testLog.message(`WS on-message, typeof ${typeof data}`);
+
+			if(typeof data == 'string') {
+				testLog.message(`WS message data: "${data}"`);
+			}
+		});
+
+		await sleep(5000);
```

Comment:
Guessing this will be removed in the non-draft version, but seeing a one-off sleeps reminds me a bit of how such an API can be misused. Consider exposing safer higher level primitives (like the polling API).

Reply:
I removed the sleep calls (more to the point, the de-activated test method it occurred in).